src/allora_sdk/worker/worker.py

```diff
@@ -69,6 +78,26 @@ class _StopQueue:
 SubmissionWindowOpenedEvent = Union[EventWorkerSubmissionWindowOpened, EventReputerSubmissionWindowOpened]
 
 
+class WorkerRole(Enum):
+    """Role of the worker in the Allora network."""
+    INFERER = "inferer"
+    REPUTER = "reputer"
+
+
+# Type aliases for reputer callbacks
+GroundTruthFnResultType = str | float
+GroundTruthFnSync = Callable[[int], GroundTruthFnResultType]
```

Comment:
It might make sense for us to have the worker framework pass the user fn not just a nonce, but also all of the worker submissions from the epoch being reputed.  What do you think?  Something like:

```python
Callable[[int, list[ValueBundle]], GroundTruthFnResultType]
```

It's needed for every reputer so it seems like an easy win (if we can get the type system to behave)